app/src/main/java/space/taran/arknavigator/mvp/model/UserPreferences.kt

```diff
@@ -39,8 +48,38 @@ class UserPreferences @Inject constructor(val context: Context) {
         else Sorting.values()[intValue]
     }
 
+    suspend fun setCrashReportEnabled(enabled: Boolean?) {
+        dataStore.edit { preferences ->
+            preferences[PreferencesKeys.CRASH_REPORT_PREFERENCE] = enabled?: BuildConfig.DEBUG
+        }
+    }
+
+    suspend fun isCrashReportEnabled(): Boolean =
+        dataStore.data.first()[PreferencesKeys.CRASH_REPORT_PREFERENCE]?: BuildConfig.DEBUG
```

Comment:
This seems to excessive since we already use it in `setCrashReportEnabled`

Reply:
This takes care of one scenario: On a fresh app, there is no saved state for `CRASH_REPORT_PREFERENCE`. When we select this option at least once or clear preferences - yes, it will be saved, and this nullability check won't be used.